renderer/components/ConsoleOutput.tsx

```diff
@@ -1,41 +1,54 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import { Panel } from 'react-bootstrap';
 
-class ConsoleOutput extends React.Component {
-  constructor(props) {
+interface StateProps {
+  toggleConsole: Function;
+  disableScroll: boolean;
+}
+
+interface OwnProps {
+  height: number;
+  output: [];
+  show: boolean;
+}
+
+type Props = StateProps & OwnProps;
+
+export class ConsoleOutput extends React.Component<Props, any> {
+  constructor(props: Props) {
     super(props);
-    this.scrollToBottom = this.scrollToBottom.bind(this);
   }
 
-  componentDidMount() {
+  componentDidMount = () => {
     this.scrollToBottom();
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (this.props.output.size === 0 && nextProps.output.size > 0 && !this.props.show) {
+  componentWillReceiveProps = (nextProps: Props) => {
+    if (this.props.output.length === 0 && nextProps.output.length > 0 && !this.props.show) {
       this.props.toggleConsole();
     }
   }
 
-  componentDidUpdate() {
+  componentDidUpdate = () => {
     this.scrollToBottom();
   }
 
-  scrollToBottom() {
+  scrollToBottom = () => {
     if (!this.props.disableScroll) {
       this.outerDiv.scrollTop = this.outerDiv.scrollHeight;
```

Comment:
is `outerDiv` a `</div>`? Because it gives `Property 'outerDiv' does not exist on type 'ConsoleOutput'` error.

Reply:
I will have to take a look at this, not sure...